lms/djangoapps/commerce/utils.py

```diff
@@ -274,16 +281,91 @@ def refund_seat(course_enrollment, change_mode=False):
             mode=course_enrollment.mode,
             user=enrollee,
         )
-        if change_mode and CourseMode.can_auto_enroll(course_id=CourseKey.from_string(course_key_str)):
-            course_enrollment.update_enrollment(mode=CourseMode.auto_enroll_mode(course_id=course_key_str),
-                                                is_active=False, skip_refund=True)
-            course_enrollment.save()
+        if change_mode:
+            _auto_enroll(course_enrollment)
     else:
         log.info('No refund opened for user [%s], course [%s]', enrollee.id, course_key_str)
 
     return refund_ids
 
 
+def _refund_in_commerce_coordinator(course_enrollment, change_mode):
+    """
+    Helper function to perform refund in Commerce Coordinator.
+
+    Parameters:
+        course_enrollment (CourseEnrollment): the enrollment to refund.
+        change_mode (bool): whether the enrollment should be auto-enrolled into
+            the default course mode after refund.
+
+    Returns:
+        bool: True if refund was performed. False if refund is not applicable
+            to Commerce Coordinator.
+    """
+    enrollment_source_system = course_enrollment.get_order_attribute_value("source_system")
+    course_key_str = str(course_enrollment.course_id)
+    # Commerce Coordinator enrollments will have an orders.source_system enrollment attribute.
+    # Redirect to Coordinator only if the source_system is whitelisted as Coordinator's in settings.
+    if enrollment_source_system and enrollment_source_system in settings.COMMERCE_COORDINATOR_REFUND_SOURCE_SYSTEMS:
+        log.info('Redirecting refund to Commerce Coordinator for user [%s], course [%s]...',
+                 course_enrollment.user_id, course_key_str)
+        # Re-use Ecommerce API client factory to build an API client for Commerce Coordinator..
+        service_user = get_user_model().objects.get(
+            username=settings.COMMERCE_COORDINATOR_SERVICE_WORKER_USERNAME
+        )
+        api_client = get_ecommerce_api_client(service_user)
+        refunds_url = urljoin(
+            settings.COMMERCE_COORDINATOR_URL_ROOT,
+            settings.COMMERCE_COORDINATOR_REFUND_PATH
+        )
+        # Build request, raising exception if Coordinator returns non-200.
+        enrollment_attributes = CourseEnrollmentAttribute.get_enrollment_attributes(course_enrollment)
+        api_client.post(
+            refunds_url,
+            json={
+                'course_id': course_key_str,
+                'username': course_enrollment.username,
+                'enrollment_attributes': enrollment_attributes
+            }
+        ).raise_for_status()
```

Comment:
@pshiu This exception raised is not being caught anywhere. If you compare it with the `_process_refund` method, it has encapsulated API call in a [try/except](https://github.com/openedx/edx-platform/pull/34438/files#diff-9d97b0582806ed85f7cd7b49ffc075dc2388097ca4b12a80b2c84a7c59d03feeR391-R403) block.


Also, do you think we need to send a notification to support in case the CC API call raises an exception (unable to process automated refunds)? i.e. https://github.com/openedx/edx-platform/pull/34438/files#diff-9d97b0582806ed85f7cd7b49ffc075dc2388097ca4b12a80b2c84a7c59d03feeR407-R428

Reply:
Okay. So, I am now recalling our workflow. This call to CC will create a return resource on CT just. We will listen to return event bus message in another endpoint. So, I think no chance we can notify support on failure.

But, yes, definitely worth ecnapsulating it in a try/except block